### PR TITLE
Twitter settings page implemented.

### PR DESCRIPTION
**Pull request by SezginEge**

### CODE REVIEW

**Comment by daronyondem** [comment]
When I tried to safe Twitter Settings with all empty boxes I got an Unhandled Exception.

Error Code: 5f49342a-70e7-4c15-842c-784ceef4dbaa

Finally, I still see msg-integration files in this PR. Are they related to the twitter settings implementation? If so how?

Thanks

**Comment by daronyondem** [comment]
I'm sorry. But you need to separate the MSG integration files from this PR. If they have shared resources feel free to submit the shared resources in separated PRs separately. Don't worry about duplication. They can be solved during the merge process.

@ilkayilknur @gokhangulbiz Can you please investigate the 5f49342a-70e7-4c15-842c-784ceef4dbaa. I'm not having it any more but I'm wondering what that error refers to. Feel free to email me this one as it is kind of off-topic at this point.

**Comment by ilkayilknur** [comment]
Thanks for reporting @daronyondem I've found and fixed the bug. I'm gonna send an internal email about  the fix.

**Comment by daronyondem** [comment]
Lovely! Thanks! @ilkayilknur

File: src/XOMNI.CMS.FrontEnd/src/pages/management/twitter-settings-page/twitter-settings.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                onlyIf: () => {
+                    if (this.consumerKey() || this.redirectUri()) {
+                        return true;
+                    }
+                    return false;
+                }
+            }
+        });
+
+        this.redirectUri.extend({
```
Comment (ilkayilknur) [inline src/XOMNI.CMS.FrontEnd/src/pages/management/twitter-settings-page/twitter-settings.ts]: We should add URI validator to redirectUri field.

File: src/XOMNI.CMS.FrontEnd/src/pages/management/twitter-settings-page/twitter-settings.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    success = (result: Models.Management.Configuration.Settings) => {
+        try {
+            this.settings = result;
+            this.consumerKey(result.TwitterConsumerKey);
+            this.consumerSecretKey(result.TwitterConsumerKeySecret);
+            this.redirectUri(result.TwitterRedirectUri);
+        }
+        catch (exception) {
```
Comment (ilkayilknur) [inline src/XOMNI.CMS.FrontEnd/src/pages/management/twitter-settings-page/twitter-settings.ts]: Why did we use so much try-catch in this page ? It looks like some of them are unnecessary. @gokhangulbiz What do you think ?

File: src/XOMNI.CMS.FrontEnd/src/pages/management/twitter-settings-page/twitter-settings.ts (hunk 1 above)
Comment (SezginEge) [inline src/XOMNI.CMS.FrontEnd/src/pages/management/twitter-settings-page/twitter-settings.ts]: Okay. I will add URI validator.

Comment (daronyondem) [inline src/XOMNI.CMS.FrontEnd/src/pages/management/twitter-settings-page/twitter-settings.ts]: Make sure we don't screw up with potential input values enabling particular development and testing scenarios. ;)

**Comment by SezginEge** [comment]
I'm closing this PR. I cleaned up and sent new PR. #62
 
@ilkayilknur I added url validation. You can checkout #60, #62.

**Pull request closed without merging**